ebu_tt_live/documents/test/test_deduplicationComparableRegions.py

```diff
@@ -0,0 +1,101 @@
+from unittest import TestCase
+import os
+from ebu_tt_live import bindings
+from ebu_tt_live.node import deduplicator
+from ebu_tt_live.documents.ebutt3 import EBUTT3Document
+from ebu_tt_live.bindings import style_type, region_type, div_type, p_type, span_type, br_type, ebuttdt
+
+
+class TestComparableElement(TestCase):
+
+    test_style1 = style_type(
+        id='SEQ58.defaultStyle1',
+        direction='ltr',
+        fontFamily='sansSerif',
+        fontSize='1c 2c',
+        lineHeight='normal',
+        textAlign='left',
+        color='rgb(255, 255, 255)',
+        backgroundColor='rgb(0, 0, 0)',
+        fontStyle='normal',
+        fontWeight='normal',
+        textDecoration='none',
+        unicodeBidi='normal',
+        wrapOption='wrap',
+        padding='5px',
+        multiRowAlign='center',
+        linePadding='0.5c'
+        )
+
+    test_style2 = style_type(
+        id='SEQ60.defaultStyle1',
+        direction='ltr',
+        fontFamily='sansSerif',
+        fontSize='1c 2c',
+        lineHeight='normal',
+        textAlign='left',
+        color='rgb(255, 255, 255)',
+        backgroundColor='rgb(0, 0, 0)',
+        fontStyle='normal',
+        fontWeight='normal',
+        textDecoration='none',
+        unicodeBidi='normal',
+        wrapOption='wrap',
+        padding='5px',
+        multiRowAlign='center',
+        linePadding='0.5c'
+        )
+
+    test_style3 = style_type(
+        id='SEQ59.defaultStyle1',
+        direction='ltr',
+        fontFamily='Serif',
+        fontSize='1c 2c',
+        lineHeight='normal',
+        textAlign='left',
+        color='rgb(0, 255, 255)',
+        backgroundColor='rgb(0, 0, 0)',
+        fontStyle='normal',
+        fontWeight='normal',
+        textDecoration='none',
+        unicodeBidi='normal',
+        wrapOption='wrap',
+        padding='5px',
+        multiRowAlign='center',
+        linePadding='0.5c'
+        )
+
+    test_style4 = style_type(
+        id='SEQ61.defaultStyle1',
+        direction='ltr',
+        fontSize='1c 2c',
+        fontFamily='Serif',
+        multiRowAlign='center',
+        lineHeight='normal',
+        textAlign='left',
+        textDecoration='none',
+        fontStyle='normal',
+        fontWeight='normal',
+        backgroundColor='rgb(0, 0, 0)',
+        unicodeBidi='normal',
+        color='rgb(0, 255, 255)',
+        wrapOption='wrap',
+        padding='5px',
+        linePadding='0.5c'
+        )
+    comparableElement = deduplicator.ComparableElement
+
+    def test_dup_elements_diff_ids(self):
+        test_instance1 = self.comparableElement(self.test_style1)
+        test_instance2 = self.comparableElement(self.test_style2)
+        assert test_instance1.my_hash == test_instance2.my_hash
```

Comment:
For all of these, should also check that the equality and non-equality operators work on the `comparableElement` objects. Are `assert_is_same` or `assert_different` available functions?

Reply:
I can't find `assert_is_same` or `assert_different`, however, the test does check the equality and non-equality operators.

They're written as different scenarios, and if I were to add the other `testinstance`s, then all the scenarios would be the same.

Reply:
The point here is that the tests only compare the _hashes_ and don't check that the `comparableElement` objects themselves have correctly functioning equality and non-equality operators.